### PR TITLE
Dev-2517 Federal Account List View

### PR DESCRIPTION
**Pull request by jonhill13**
**High level description:**

Add Federal Account List View to Federal Accounts Section.

**Technical details:**

* create federal accounts table

* add state for table to AwardMetaDataContainer

* add sort and page methods

* add parse awards method

* add method for fed accounts to idv helper

* add mock data for fed accounts idv helper

* add fed account summary model

* add methods to model to handle account name and percentages

* add mapping data for fed account table

* add tests for award metadata container and model

* change column display name for award history section fed account funding table

* change name for row in fed account summary table

* update table css

**JIRA Ticket:**
[DEV-2517](https://federal-spending-transparency.atlassian.net/browse/DEV-2517)

**Mockup:**
https://bahdigital.invisionapp.com/share/69IA8EPGPCM#/screens/296008376

The following are ALL required for the PR to be merged:
- [ ] Code review
- [x] Design review
- [x] [API contract](https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/api_contracts/contracts/award/idv/FundingAccounts.md)
- [x] [API #1783](https://github.com/fedspendingtransparency/usaspending-api/pull/1783) 
- [x] Verified cross-browser compatibility

### CODE REVIEW

**Comment by AGuyNamedMarco** [comment]
Screen Shot 2019-05-06 at 11 52 55 AM

Thanks @jonhill13 .

Notes (screenshot above:
- The pagination should be fixed in its position at the bottom-right corner.  Right now, it scrolls with the content.
- When clicking through the pagination, they content quickly collapses its height, then lengthening once the content loads.  It's preferable to maintain the height of this section during the transition if possible.
- When no data is available, can we display the following design (screen below)?

federalAccounts--noData

File: src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    inFlight,
+    error,
+    page,
+    limit,
+    sort,
+    order,
+    total,
+    federalAccounts,
+    changePage,
+    updateSort
```
Comment (MaxwellKendall) [inline src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx]: These are a lot of props for this component. Makes it hard to read at a glance. I would recommend moving the FederalAccountsTable outside of this component, and putting it into the parent component. 

So then "AwardMetaDataContainer" would render two seaparate components.

File: src/_scss/pages/awardV2/idv/_awardFundingSummary.scss [hunk 2]
```diff
@@ -1,21 +1,36 @@
...
+          .loading-animation {
+              @import "../../search/results/screens/loading";
+          }
+      }
     }
-    .federal-accounts__section {
-        @extend %coming-soon-idv;
-        height: 200px;
+  }
+  .federal-accounts-summary__section {
```
Comment (MaxwellKendall) [inline src/_scss/pages/awardV2/idv/_awardFundingSummary.scss]: In the same way this CSS is broken out so `.federal-accounts__section` is a sibling of `.feder-accounts-summary__section` I recommend having two sibling components in the `AwardMetaDataContainer` so you see "Ok, this container has two children..."

File: src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx (hunk 1 above)
Comment (jonhill13) [reply src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx]: The tree map PR creates a FederalAccountsSection class that handles rendering the Toggle Buttons, Error/Loading/No Results Messages, TreeMap, and Table

**Comment by jonhill13** [comment]
@AGuyNamedMarco Thank you for taking a look.

- [x] The pagination should be fixed in its position at the bottom-right corner. Right now, it scrolls with the content.

- [ ] When clicking through the pagination, they content quickly collapses its height, then lengthening once the content loads. It's preferable to maintain the height of this section during the transition if possible.

- [ ] When no data is available, can we display the following design (screen below)?
This is what I have currently, will this suffice?
image

**Comment by jonhill13** [comment]
@AGuyNamedMarco please check out the above comments

File: src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx (hunk 1 above)
Comment (jonhill13) [reply src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx]: TreeMap PR has been updated from a FederalAccountsSummary class to component

File: src/_scss/pages/awardV2/idv/_awardFundingSummary.scss (hunk 2 above)
Comment (jonhill13) [reply src/_scss/pages/awardV2/idv/_awardFundingSummary.scss]: Cool, currently, in the TreeMap PR the overarching class is `.federal-accounts`,  which is broken into `.federal-accounts__section` and `.federal-accounts-summary__section`

File: src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx (hunk 1 above)
Comment (jonhill13) [reply src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx]: @MaxwellKendall

File: src/_scss/pages/awardV2/idv/_awardFundingSummary.scss (hunk 2 above)
Comment (jonhill13) [reply src/_scss/pages/awardV2/idv/_awardFundingSummary.scss]: @MaxwellKendall

**Pull request merged**